src/Microsoft.OData.Core/Microsoft.OData.Core.txt

```diff
@@ -842,10 +842,15 @@ CustomUriFunctions_AddCustomUriFunction_CustomFunctionOverloadExists=The given f
 RequestUriProcessor_InvalidValueForEntitySegment=The ODataPathSegment provided (Id = {0}) is not an EntitySetSegment.
 RequestUriProcessor_InvalidValueForKeySegment=The KeySegment provided (Id = {0}) is either null, having no keys, or does not target a single resource.
 
+RequestUriProcessor_CannotApplyFilterOnSingleEntities=$filter path segment cannot be applied on single entities or singletons. Entity type: '{0}'.
+RequestUriProcessor_NoNavigationSourceFound=There are no navigation sources found to apply '{0}'.
+RequestUriProcessor_FilterPathSegmentRequiresParameterAlias=The $filter path segment must contain a single parameter alias in the form $filter=@alias.
+
 ; Copied from the server, will be incrementally replaced.
 RequestUriProcessor_EmptySegmentInRequestUrl=Empty segment encountered in request URL. Please make sure that a valid request URL is specified.
 RequestUriProcessor_SyntaxError=Bad Request - Error in query syntax.
 RequestUriProcessor_CountOnRoot=The request URI is not valid, the segment $count cannot be applied to the root of the service.
+RequestUriProcessor_FilterOnRoot=The request URI is not valid, the segment $filter cannot be applied to the root of the service.

```

Comment:
>segment $filter [](start = 67, length = 15)

..., the $filter path segment ...

Reply:
I am going to keep as-is to make it consistent with the RequestUriProcessor_CountOnRoot message. I think it's clear enough that it's a path segment.